pipeline/custom.groovy

```diff
@@ -0,0 +1,162 @@
+import groovy.io.FileType
+def nodeName = "centos-7"
+def sharedLib
+
+def GetCLI(args){
+	/*
+		Generates the CLI using the arguments provided and returns it.
+	*/
+	echo "Inside get cli"
+	def cli = 'cd $WORKSPACE; .venv/bin/python run.py'
+	cli += ' --rhbuild ' + args['rhbuild']
+	cli += ' --platform ' + args['platform']
+	cli += ' --build ' + args['build']
+	cli += ' --v2'
+	cli += ' --global-conf ' + args['global_conf']
+	cli += ' --osp-cred ~/osp-cred-ci-2.yaml'
+	cli += ' --inventory ' + args['inventory']
+	cli += ' --suite ' + args['suite_name']
+	cli += ' --instances-name ' + args['instances_name']
+	cli += ' ' + args['addn_args']
+
+	if(cli.indexOf('log-level') < 0){
+	    cli += ' --log-level info'
+	}
+
+	if(args['rhs_ceph_repo']?.trim() || args['container_image']?.trim()){
+		cli += ' --ignore-latest-container'
+	}
+	if(args['rhs_ceph_repo']?.trim()){
+		cli += ' --rhs-ceph-repo ' + args['rhs_ceph_repo']
+	}
+	if(args['add_repo']?.trim()){
+		cli += ' --add-repo ' + args['add_repo']
+	}
+	if(args['hotfix_repo']?.trim()){
+		cli += ' --hotfix-repo ' + args['hotfix_repo']
+	}
+	if(args['container_image']?.trim()){
+		container = args['container_image'].split('.com/')
+		docker_registry = container[0] + '.com'
+		docker = container[1].split(':')
+		docker_tag = docker.last()
+		docker_image = docker.first()
+		cli += ' --insecure-registry'
+		cli += ' --docker-registry ' + docker_registry
+		cli += ' --docker-tag ' + docker_tag
+		cli += ' --docker-image ' + docker_image
+	}
+	echo "cli"
+	echo cli
+	return cli
+}
+
+def ExecuteBuildStage(instances_name){
+    echo "Inside execute build stages"
+    Ceph_Version = 'pacific'
+    if("3." in "${params.RHCS_Build}"){
+        Ceph_Version = 'luminous'
+    }
+    else if("4." in "${params.RHCS_Build}"){
+        Ceph_Version = 'nautilus'
+    }
+
+    def args = [
+		'rhbuild' : "${params.RHCS_Build}",
+		'platform' : "${params.Platform}",
+		'build' : "${params.Build}",
+		'inventory' : "conf/inventory/${params.Inventory}.yaml",
+		'rhs_ceph_repo' : "${params.Override_Ceph_Base_URL}",
+		'container_image' : "${params.Override_Container_Image}",
+		'add_repo' : "${params.Additional_Repo_File}",
+		'hotfix_repo' : "${params.Hotfix_Repo_File}",
+		'suite_name' : "suites/${Ceph_Version}/${params.Group}/${params.Suite}.yaml",
+		'global_conf' : "conf/${Ceph_Version}/${params.Group}/${params.Conf}.yaml",
+		'instances_name' : instances_name,
+		'addn_args': "${params.Additional_Args}"
+	]
+    def destroy_cluster = "${params.Destroy_Cluster}"
+    def cli = GetCLI(args)
+    String stageName = "Stage ${params.Suite}"
+    stage(stageName){
+        def dirName = UUID.randomUUID().toString()
```

Comment:
Can we execute in the current workspace?

Reply:
Done.